cmd/spire-agent/cli/run/run_posix_test.go

```diff
@@ -5,15 +5,121 @@ package run
 
 import (
 	"bytes"
+	"fmt"
+	"io"
 	"os"
+	"syscall"
 	"testing"
 
 	"github.com/hashicorp/hcl/hcl/printer"
 	"github.com/spiffe/spire/pkg/agent"
+	commoncli "github.com/spiffe/spire/pkg/common/cli"
+	"github.com/spiffe/spire/pkg/common/log"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+func TestCommand_Run(t *testing.T) {
+	testTempDir := t.TempDir()
+	testDataDir := fmt.Sprintf("%s/data", testTempDir)
+	testAgentSocketDir := fmt.Sprintf("%s/spire-agent", testTempDir)
+
+	type fields struct {
+		logOptions         []log.Option
+		env                *commoncli.Env
+		allowUnknownConfig bool
+	}
+	type args struct {
+		args []string
+	}
+	type want struct {
+		code               int
+		dataDirCreated     bool
+		agentUdsDirCreated bool
+	}
+	tests := []struct {
+		name     string
+		osTarget string
+		fields   fields
+		args     args
+		want     want
+	}{
+		{
+			name:     "don't create any dir when error loading config",
+			osTarget: "linux",
+			args: args{
+				args: []string{},
+			},
+			fields: fields{
+				logOptions: []log.Option{},
+				env: &commoncli.Env{
+					Stderr: io.Discard,
```

Comment:
I think that it would be good to test the content of Stderr, so we can assert the reason of the failure. We may have a `*bytes.Buffer` var here that you can then convert to a string for the comparison.
There is an example of how this can be done here: https://github.com/spiffe/spire/blob/main/cmd/spire-agent/cli/healthcheck/healthcheck_test.go#L67